src/components/visualization/perspectiveViz/perspectiveViz.tsx

```diff
@@ -184,35 +189,40 @@ function PerspectiveViz(props: TableData) {
   const transformedConfig = props.config && transformTableConfig(props.config)
 
   useEffect(() => {
-    const worker = perspective.worker()
-    worker
-      .table(transformTableData(props.data, props.headers))
-      .then((table) => {
-        const viewer = viewerRef.current
-        if (viewer) {
-          viewer
-            .load(table)
-            .then(() => {
-              viewer.restore({
-                ...transformedConfig,
-                theme: perspectiveTheme,
-                title: props.title,
-                settings: false,
+    const loadTable = () => {
+      const viewer = viewerRef.current
+      if (viewer) {
+        perspective
+          .worker()
+          .then((worker: Client) =>
+            worker.table(transformTableData(props.data, props.headers))
+          )
+          .then((table: Table) => {
+            viewer
+              .load(table)
+              .then(() =>
+                viewer.restore({
+                  ...transformedConfig,
+                  theme: perspectiveTheme,
+                  title: props.title,
+                  settings: false,
+                })
+              )
+              .then(() => {
+                if (viewer.shadowRoot) {
+                  const sheet = new CSSStyleSheet()
+                  sheet.replaceSync(perspectiveVizAdditionalStyles)
+                  viewer.shadowRoot.adoptedStyleSheets.push(sheet)
+                }
               })
-              if (viewer.shadowRoot) {
-                const sheet = new CSSStyleSheet()
-                sheet.replaceSync(perspectiveVizAdditionalStyles)
-                viewer.shadowRoot.adoptedStyleSheets.push(sheet)
-              }
-            })
-            .catch(() => {
-              setHasError(true)
-            })
-        }
-      })
-      .catch(() => {
-        setHasError(true)
-      })
+          })
+          .catch(() => {
+            setHasError(true)
+          })
+      }
+    }
+
+    loadTable()
```

Comment:
I think this could be simplified,

```typescript
  useEffect(() => {
    const viewer: HTMLPerspectiveViewerElement = viewerRef.current
    if (viewer) {
      perspective
        .worker()
        .then((worker: Client) => {
          worker.table(transformTableData(props.data, props.headers)).then((table: Table) => {
            viewer.load(table).then(() =>
              viewer.restore({
                ...transformedConfig,
                theme: perspectiveTheme,
                title: props.title,
                settings: false,
              }).then(() => {
                if (viewer.shadowRoot) {
                  const sheet = new CSSStyleSheet()
                  sheet.replaceSync(perspectiveVizAdditionalStyles)
                  viewer.shadowRoot.adoptedStyleSheets.push(sheet)
                }
              }))
          })
        }).catch(() => {
        setHasError(true)
      })
    }
  }, [props.data, perspectiveTheme])
  ```